modin/pandas/dataframe.py

```diff
@@ -108,6 +109,20 @@ def __init__(
         query_compiler=None,
     ):
         Engine.subscribe(_update_engine)
+
+        if data is not None and hasattr(data, "__dataframe__"):
+            if StorageFormat.get() == "Pandas":
+                from modin.core.dataframe.pandas.dataframe.protocol.utils import (
+                    from_dataframe,
+                )
+                from modin.core.storage_formats.pandas.query_compiler import (
+                    PandasQueryCompiler,
+                )
```

Comment:
this probably should be a part of `FactoryDispatcher`

Reply:
fixed